src/hostname/hostnamectl.c

```diff
@@ -257,6 +282,21 @@ static int set_hostname(int argc, char **argv, void *userdata) {
         if (!arg_pretty && !arg_static && !arg_transient)
                 arg_pretty = arg_static = arg_transient = implicit = true;
 
+        if (arg_transient) {
+                _cleanup_free_ char *source = NULL;
+
+                r = get_one_name(bus, "HostnameSource", &source);
+                if (r < 0)
+                        return r;
+
+                if (streq(source, hostname_source_to_string(HOSTNAME_STATIC))) {
```

Comment:
wouldn't it be more natural to do the comparison by enum rather than by string. i.e. `hostname_source_from_string(source) == HOSTNAME_STATIC`?